### DISCUSSION TITLE
USB keyboard not detected

### DISCUSSION THREAD

**Discussion by genki24** [hrvach/deskhop#194] [Q&A]
Hello,
first of all let me thank for this excellent work! I tried to make it work on an "Adafruit Feather RP2040 with USB Type A Host", which -however- has different GPIO usage (USB on GPIO 16 & 17, UART on GPIO 12 & 13 for both devices). Thought a simple code adjustment (#define PIO_USB_DP_PIN 16, #define BOARD_..._RX 13, #define BOARD_..._TX 12) should to the trick, but ran into trouble. The LED never blinks, so the keyboard is never detected correctly. Hardware seems ok: Checked wiring several times, all voltages within limits, hid-remapper firmare detects keyboard and runs as intended. I'm running out of ideas? Any help appreciated!

**Comment by hrvach** [comment]
I'll try taking a look in a few days once I'm back behind a computer. Apologies for the delay and all the best for the holidays!

**Comment by genki24** [comment]
No worries! Some hand-wiring to default pins brought it alive as expected. Happy so far. Indeed there might be conflicting/double defaults (CMakeList.txt : set(DP_PIN_DEFAULT 14 CACHE STRING  "Default USB D+ Pin Number") vs. main.h : #define PIO_USB_DP_PIN 14). Will further investigate.

**Reply by hrvach** [reply]
Thanks for finding it, let me know if you manage to figure it out in the meantime!
